scripts/views/mapView.js

```diff
@@ -1,439 +1,391 @@
+  function highlightDistrict (geoid) {
+    var visibility = map.getLayoutProperty('selected-fill', 'visibility');
+    // If there are no selections, turn dem layers on
+    if (visibility === 'none') {
+      map.setLayoutProperty('selected-fill', 'visibility', 'visible');
+      map.setLayoutProperty('selected-border', 'visibility', 'visible');
+    }
+
+    // Filter for which district has been selected
+    var filter = ['all', ['==', 'GEOID', geoid]];
+
+    // Set that layer filter to the selected
+    map.setFilter('selected-fill', filter);
+    map.setFilter('selected-border', filter);
+  }
+
+  // Make the Table below the map
+  function makeTable (data) {
+    var tableRowTemplate = Handlebars.getTemplate('eventTableRow');
+    for (key in data){
+      var ele = data[key];
+      if(ele.Member){
+        var id = ele.Member+ele.Date;
+        ele.rowid = id.replace(/[\W]/g, '');
+        TownHall.allTownHalls.push(ele);
+        $('#all-events-table').append(tableRowTemplate(ele));
+      }
+    }
+  }
+
+  // Assign Zip Lookup function to Search field
+  function sidebarEvents () {
+    $('#look-up').on('submit', eventHandler.lookup);
+  }
+
+  // Zip Code Lookup!
+  eventHandler.lookup = function (e) {
+    e.preventDefault();
+    var zip = $('#look-up input').val();
+    if (zip) {
+      var validDistricts = [];
+      var callbackTrigger = 0;
+      var thisState;
+
+      zipLookup.forEach(function(n){
+        if (n.zip === zip) {
+          focusMap(n.abr, n.dis);
+          thisState = n.abr;
+          validDistricts.push(n.dis);
+        // Fall back on first four strings if exact zip match doesn't exist
+        } else if (n.zip.substring(0,4) === zip.substring(0,4)) {
+          focusMap(n.abr, n.dis);
+          thisState = n.abr;
+          validDistricts.push(n.dis);
+        }
+
+        callbackTrigger++;
+        if(callbackTrigger === zipLookup.length){
+          makeSidebar(thisState, validDistricts);
+        }
       });
-    });
+    }
   };
 
-  readData();
+  // Match the looked up zip code to district #
+  function matchSelectionToZip (state, districts) {
+    var fetchedData = []
+
+    for (key in townhallData){
+      var k = townhallData[key];
+      if (k.StateAb === state) {
+        if (k.District === 'Senate') {
+          fetchedData.push(k)
+        } else {
+          if(districts.constructor === Array) {
+            districts.forEach((d) => {
+              let districtMatcher = state + '-' + d
+              if (k.District === districtMatcher) {
+                fetchedData.push(k)
+              }
+            })   
+          } else {
+            let districtMatcher = state + '-' + districts
+            if (k.District === districtMatcher) {
+              fetchedData.push(k)
+            }
+          }
+        }
+      }
+    }
+
+    return fetchedData
+  }
+
+  // Create a sidebar and load it with Town Hall event cards.
+  function makeSidebar (state, districts) {
```

Comment:
Fixed. Was having issues due to the districts being referred to as '01' or '1' inconsistently throughout the data.